### PR TITLE
bpf: Do not delete and re-add ip rules and routes in the proxy table

### PR DESCRIPTION
**Pull request by tgraf**
This can otherwise lead to a brief interruption of traffic from the proxy to
local endpoints when the agent is being restarted.

Related: #5651

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5653)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

green build

File: bpf/init.sh [hunk 1]
```diff
@@ -140,19 +125,15 @@ function move_local_rules()
...
-	delete_old_ip_rules $PROXY_RT_TABLE
+	# Any packet from a local process uses a separate routing table
+	rulespec="fwmark 0xA00/0xF00 pref 10 lookup $PROXY_RT_TABLE"
 
-	if [ -n "$(ip -4 rule list)" ]; then
-		ip -4 route flush table $PROXY_RT_TABLE
-		# Any packet from a proxy uses a separate routing table
-		ip -4 rule add fwmark 0xA00/0xF00 pref 10 lookup $PROXY_RT_TABLE
+	if [ -z "$(ip -4 rule list $rulespec)" ]; then
+		ip -4 rule add $rulespec
```
Comment (joestringer) [inline bpf/init.sh:132]: Do these need to be `|| true` to handle `EEXIST`?

```
$ sudo ip rule add fwmark 0xA00/0xF00 pref 10 lookup 2005
$ echo $?
0
$ sudo ip rule add fwmark 0xA00/0xF00 pref 10 lookup 2005
RTNETLINK answers: File exists
$ echo $?
2
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Seems like these scripts will exit early / fail out if the ip rules already exist.

File: bpf/init.sh (hunk 1 above)
Comment (tgraf) [reply bpf/init.sh:132]: That's why it checks if the rule already exists with `ip rule list`. Unfortunately there is no `ip rule replace`.

**Review by joestringer — APPROVED** [review]

**Comment by tgraf** [comment]
Will rebase after #5663 is merged

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**